src/python/pants/backend/python/lint/docformatter/rules.py

```diff
@@ -58,23 +59,40 @@ def generate_args(
 
 @rule(level=LogLevel.DEBUG)
 async def setup_docformatter(setup_request: SetupRequest, docformatter: Docformatter) -> Setup:
-    docformatter_pex_request = Get(
+    if docformatter.lockfile == "<none>":
+        requirements = PexRequirements(docformatter.all_requirements)
+    elif docformatter.lockfile == "<default>":
+        requirements = PexRequirements(
+            file_content=FileContent(
+                "docformatter_default_lockfile.txt",
+                importlib.resources.read_binary(
+                    "pants.backend.python.lint.docformatter", "lockfile.txt"
+                ),
+            )
+        )
+    else:
+        requirements = PexRequirements(
+            file_path=docformatter.lockfile,
+            file_path_description_of_origin="the option `[docformatter].experimental_lockfile`",
+        )
```

Comment:
Rather than being something that a tool should explicitly request, this feels like an argument that should be passed down via `PexRequest`. Maybe a `Resolve(name=.., file=.., default=..)` parameter, which `PexRequest` would switch on behavior with? An end-user resolve would have no default, etc.